### PR TITLE
Normalize search

### LINKED ISSUE #14: Normalize use of search in models
All our search methods are similar. We should normalize this.

### PR DESCRIPTION
**Pull request by jimmypoulsen**
Fixes #14

### CODE REVIEW

File: lib/synsbasen_api/resources/version.rb [hunk 1]
```diff
@@ -20,7 +20,13 @@ def find(id, expand: [])
       # @return [ApiResponse] An instance of `ApiResponse` containing details
       #   of all versions associated with the specified variant.
       def all(variant_id, expand: [])
-        get("/v1/variants/#{variant_id}/versions", expand: expand)
+        get("/v1/#{resource_name}/#{variant_id}/versions", expand: expand)
```
Comment (Tobias-Knudsen) [inline lib/synsbasen_api/resources/version.rb:23]: This doesn't seem correct - /versions should be on the /variants/:id/versions endpoint

```suggestion
        get("/v1/variants/#{variant_id}/#{resource_name}", expand: expand)
```

File: lib/synsbasen_api/resources/version.rb [hunk 2]
```diff
@@ -3,15 +3,15 @@
...
     class << self
       # Retrieves information about a specific version based on its ID.
       #
       # @param id [String] The unique identifier of the version.
       # @return [ApiResponse] An instance of `ApiResponse` containing details
       #   of the specified version.
       def find(id, expand: [])
-        get("/v1/versions/#{id}", expand: expand)
+        get("/v1/#{resource_name}/#{id}", expand: expand)
       end
```
Comment (Tobias-Knudsen) [inline lib/synsbasen_api/resources/version.rb:15]: All the finds are identical at this point too. Perhaps we need a findable module too?

**Review by Tobias-Knudsen — CHANGES_REQUESTED** [review]
Vari nice 👍🏻

File: lib/synsbasen_api/resources/version.rb (hunk 2 above)
Comment (jimmypoulsen) [reply lib/synsbasen_api/resources/version.rb]: Great idea! Implemented.

**Review by Tobias-Knudsen — APPROVED** [review]
🔥

**Pull request merged**